### PR TITLE
Migrate scripts to makefile

### PR DESCRIPTION
**Pull request by skitt**

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -81,7 +82,31 @@ generate-embeddedyamls: pkg/subctl/operator/common/embeddedyamls/yamls.go
 pkg/subctl/operator/common/embeddedyamls/yamls.go: pkg/subctl/operator/common/embeddedyamls/generators/yamls2go.go $(shell find deploy/ -name "*.yaml") vendor/modules.txt
 	go generate pkg/subctl/operator/common/embeddedyamls/generate.go
 
-.PHONY: $(TARGETS) test validate build ci clean generate-embeddedyamls operator-image
+# genclientset generates the clientset for the Submariner APIs
+# It needs to be run when the Submariner APIs change
+genclientset:
```
Comment (tpantelis) [inline Makefile:87]:
```suggestion
generate-clientset:
```

To be consistent with other target names.

File: Makefile [hunk 2]
```diff
@@ -81,7 +82,31 @@ generate-embeddedyamls: pkg/subctl/operator/common/embeddedyamls/yamls.go
...
+		github.com/submariner-io/submariner-operator/pkg/apis \
+		submariner:v1alpha1
+
+# generate-operator-api updates the generated operator code
+# It needs to be run when the CRDs or APIs change
+generate-operator-api:
+	operator-sdk generate k8s
+	operator-sdk generate openapi
+
+preload_images:
```
Comment (tpantelis) [inline Makefile:101]:
```suggestion
preload-images:
```

To be consistent...

File: Makefile (hunk 1 above)
Comment (skitt) [reply Makefile:87]: Right, I wanted to keep the names as-is in case anything else referred to them, but it seems nothing does.

File: Makefile [hunk 3]
```diff
@@ -69,9 +71,8 @@ bin/subctl-%: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg
 	GOOS=$${components[-2]}; \
 	GOARCH=$${components[-1]}; \
 	export GOARCH GOOS; \
-	source $(SCRIPTS_DIR)/lib/version; \
 	$(SCRIPTS_DIR)/compile.sh \
-		--ldflags "-X github.com/submariner-io/submariner-operator/pkg/version.Version=$${VERSION}" \
+		--ldflags "-X github.com/submariner-io/submariner-operator/pkg/version.Version=$(VERSION)" \
```
Comment (mangelajo) [inline Makefile:75]: Wait, you are reverting the changes that generate subctl-devel- binaries for bleeding edge.

File: Makefile [hunk 4]
```diff
@@ -69,9 +71,8 @@ bin/subctl-%: pkg/subctl/operator/common/embeddedyamls/yamls.go $(shell find pkg
 	GOOS=$${components[-2]}; \
 	GOARCH=$${components[-1]}; \
 	export GOARCH GOOS; \
-	source $(SCRIPTS_DIR)/lib/version; \
```
Comment (mangelajo) [inline Makefile:72]: Don't revert this.

File: Makefile (hunk 3 above)
Comment (mangelajo) [reply Makefile:75]: Please see: https://github.com/submariner-io/submariner-operator/pull/522/files

Comment (skitt) [reply Makefile]: Good catch, thanks! I’ve updated the patch to use more explicit variable names in this instance.

File: Makefile [hunk 5]
```diff
@@ -81,7 +84,31 @@ generate-embeddedyamls: pkg/subctl/operator/common/embeddedyamls/yamls.go
 pkg/subctl/operator/common/embeddedyamls/yamls.go: pkg/subctl/operator/common/embeddedyamls/generators/yamls2go.go $(shell find deploy/ -name "*.yaml") vendor/modules.txt
 	go generate pkg/subctl/operator/common/embeddedyamls/generate.go
 
-.PHONY: $(TARGETS) test validate build ci clean generate-embeddedyamls operator-image
+# generate-clientset generates the clientset for the Submariner APIs
+# It needs to be run when the Submariner APIs change
+generate-clientset:
+	go get k8s.io/code-generator@kubernetes-1.14.1
+	$${GOPATH}/pkg/k8s.io/code-generator@v0.0.0-20190311093542-50b561225d70/generate-groups.sh \
```
Comment (mkolesnik) [inline Makefile:91]:
```suggestion
	$${GOPATH}/pkg/mod/k8s.io/code-generator@v0.0.0-20190311093542-50b561225d70/generate-groups.sh \
```

Shouldn't this be the path?

I ran `go get k8s.io/code-generator@kubernetes-1.14.1` inside `make shell` and see that it's inside the `mod` directory

File: Makefile [hunk 6]
```diff
@@ -81,7 +84,31 @@ generate-embeddedyamls: pkg/subctl/operator/common/embeddedyamls/yamls.go
...
+	operator-sdk generate k8s
+	operator-sdk generate openapi
+
+preload-images:
+	source $(SCRIPTS_DIR)/lib/debug_functions; \
+	source $(SCRIPTS_DIR)/lib/deploy_funcs; \
+	set -e; \
+	for image in submariner submariner-route-agent submariner-operator lighthouse-agent submariner-globalnet lighthouse-coredns; do \
+		import_image quay.io/submariner/$${image}; \
+	done
```
Comment (mkolesnik) [inline Makefile:109]: This one I'd actually leave in place, but whatever you think makes sense..

File: Makefile (hunk 5 above)
Comment (skitt) [reply Makefile]: It should, and I even had that fixed locally, but forgot to commit it...

File: Makefile (hunk 6 above)
Comment (skitt) [reply Makefile:109]: I quite like having `scripts` pretty much empty ;-).

**Pull request merged**